internal/dataplane/parser/translators/httproute.go

```diff
@@ -350,14 +352,48 @@ func mustMarshalJSON[T any](val T) string {
 	return string(key)
 }
 
-// GeneratePluginsFromHTTPRouteFilters converts HTTPRouteFilter into Kong plugins.
+// SetRoutePlugins converts HTTPRouteFilter into Kong plugins. The plugins are set into the given kongstate.Route.
+// The plugins can be set in two different ways:
+// - Direct conversion from the respective HTTPRouteFilter.
+// - ExtensionRef to plugins annotation from the ExtensionRef filter.
+func SetRoutePlugins(route *kongstate.Route, filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) error {
+	plugins, pluginAnnotation, err := generatePluginsFromHTTPRouteFilters(filters, path, tags)
+	if err != nil {
+		return err
+	}
+	if len(plugins) > 0 {
+		if route.Plugins == nil {
+			route.Plugins = make([]kong.Plugin, 0)
+		}
+		route.Plugins = append(route.Plugins, plugins...)
+	}
```

Comment:
```suggestion
		route.Plugins = append(route.Plugins, plugins...)
	}
```
appending to `nil` slice is safe